From ed1532afa37b20921a6558a019aacbf4ca38a093 Mon Sep 17 00:00:00 2001
From: Gyorgy Sarvari <skandigraun@gmail.com>
Date: Fri, 10 Jan 2025 15:56:35 +0100
Subject: [PATCH] Add new variables to u-boot environment: bootpart: the
 partition number where the boot partition resides (containing the kernel
 image to load) first_boot: used to save the environment to the persistent
 storage upon the very first boot of the system save_init_env: script to save
 the environment to persistent storage update_bootpart: check if an update was
 done recently (by verifying the value/existence of 'update_done' variable. If
 it is true, then switch the bootpartition to boot from.

Upstream-Status: Inappropriate
---
 include/configs/rk3399_common.h | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/include/configs/rk3399_common.h b/include/configs/rk3399_common.h
index c5bcd7dc5e..40397d390c 100644
--- a/include/configs/rk3399_common.h
+++ b/include/configs/rk3399_common.h
@@ -36,7 +36,10 @@
 	"partitions=" PARTS_DEFAULT	\
 	ENV_MEM_LAYOUT_SETTINGS		\
 	ROCKCHIP_DEVICE_SETTINGS	\
-	"boot_targets=" BOOT_TARGETS "\0"
+	"boot_targets=" BOOT_TARGETS "\0" \
+        "bootpart=1\0" \
+        "first_boot=1\0" \
+        "save_init_env=if itest.s x$first_boot != x; then env delete first_boot; saveenv; fi\0"
 
 #endif /* CONFIG_XPL_BUILD */
 
-- 
2.47.1

